File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.js
function solution(board) {
    let maxSide = 0;
    
    const rows = board.length;
    const cols = board[0].length;

    for (let i = 0; i < board.length; i++) {
        if (board[i][0] === 1) maxSide = 1;
    }
    for (let j = 0; j < board[0].length; j++) {
        if (board[0][j] === 1) maxSide = 1;
    }

    // 2. DP 루프 (1부터 시작!)
    for (let i = 1; i < rows; i++) {
        for (let j = 1; j < cols; j++) {
            if (board[i][j] === 1) {
                board[i][j] = Math.min(board[i - 1][j], board[i][j - 1], board[i - 1][j - 1]) + 1;
                
                if (board[i][j] > maxSide) maxSide = board[i][j];
            }
        }
    }

    return maxSide * maxSide;
}

File: 프로그래머스/2/12905. 가장 큰 정사각형 찾기/가장 큰 정사각형 찾기.py
def solution(board):
    row_count = len(board)
    col_count = len(board[0])
    
    max_side = 0
    for row in board:
        if max(row) > 0:
            max_side = 1
            break

    for i in range(1, row_count):
        for j in range(1, col_count):
            if board[i][j] == 1:
                board[i][j] = min(board[i-1][j], board[i][j-1], board[i-1][j-1]) + 1
                if board[i][j] > max_side:
                    max_side = board[i][j]
    
    return max_side * max_side